codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsProtocolUtils.java

```diff
@@ -223,6 +223,20 @@ static void generateHttpProtocolTests(GenerationContext context) {
         ).generateProtocolTests();
     }
 
+    public static void writeAwsQueryErrorCodeDeserializer(GenerationContext context) {
+        GoWriter writer = context.getWriter().get();
+        writer.write("queryCodeHeader := response.Header.Get(\"x-amzn-query-error\")");
+        writer.write("if queryCodeHeader != \"\" {");
```

Comment:
`if` statements can use the `openBlock` method on the writer to automatically handle indentation.